src/ts/app/editor/src/reducer.ts

```diff
@@ -0,0 +1,29 @@
+import {ImprovedLoop, ImprovedLoopReducer} from "app/framework"
+import {storageAction} from "app/storage"
+import {Subtree} from "./subtree"
+import * as stateTransition from "./state-transition"
+
+type HandledAction = storageAction.AddedPythonModule
+
+export interface ReducerInputs {
+    improvedLoop: ImprovedLoop<Subtree, HandledAction, any>
+}
+
+export function createReducer(inputs: ReducerInputs): ImprovedLoopReducer<Subtree, HandledAction, any> {
+    return function reduce(_previous: Subtree, action: HandledAction): Subtree | [Subtree, HandledAction] {
+        const previous = stateTransition.bumpUpdateCounter(_previous)
+
+        switch (action.type) {
+
+            case storageAction.Keys.ADDED_PYTHON_MODULE:
+                return stateTransition.setNextEditorContent(
+                    previous,
+                    action.documentCollection.documents[action.documentCollection.documents.length - 1])
+
+            default:
+                return previous
```

Comment:
do you want to do the never trick here? or don't worry about unhandled actions?

Reply:
Yeah check it out now. I did an end-around on the typescript compiler, to try and leverage it for "local" action-type-union exhaustiveness checking, but at runtime, make it so any action can pass through the code.

(Warning: it's a little weird!)